tests/test_evaluation.py

```diff
@@ -0,0 +1,141 @@
+# MIT License
+#
+# Copyright (c) 2021-23 Tskit Developers
+# Copyright (c) 2020-21 University of Oxford
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+"""
+Test tools for mapping between node sets of different tree sequences
+"""
+from collections import defaultdict
+
+import msprime
+import numpy as np
+import pytest
+import scipy.sparse
+import tsinfer
+
+from tsdate import evaluation
+
+# --- simulate test case ---
+true_ts = msprime.sim_ancestry(
+    samples=10,
+    sequence_length=1e5,
+    population_size=1e4,
+    recombination_rate=1e-8,
+    random_seed=1024,
+)
+true_ts = msprime.sim_mutations(true_ts, rate=2e-8, random_seed=1024)
```

Comment:
perhaps assert that the example has >1 tree? (I see that it ought to, but to make sure...)

Reply:
A nice way of testing things would be to run the simulation with a bunch of CensusEvents, to insert unary nodes, then check that when you compare a simplified version of this tree sequence to the original one, that everything matches exactly (eg times). And that the right thing happens when comparing the other way.